src/modules/campaign/ContentfulClient.spec.ts

```diff
@@ -1,87 +1,273 @@
-          {
-            headers: {
-              Authorization: `Bearer ${mockToken}`
-            }
+        expect(result).toEqual({
+          title: {
+            [ContentfulLocale.enUS]: 'English Title',
+            [ContentfulLocale.es]: 'Título en Español',
+            [ContentfulLocale.zh]: '中文标题'
           }
-        )
+        })
+      })
+    })
+
+    describe('when the request fails', () => {
+      beforeEach(() => {
+        nock(CMS_URL)
+          .get(
+            `/spaces/${mockSpace}/environments/${mockEnvironment}/entries/${mockId}/`
+          )
+          .query({ locale: ContentfulLocale.enUS })
+          .reply(500)
       })
 
-      it('should return the parsed response', async () => {
-        const result = await client.fetchEntry(
+      it('should throw an error', async () => {
+        await expect(
+          client.fetchEntryAllLocales(mockSpace, mockEnvironment, mockId)
+        ).rejects.toThrow('Error fetching entry in all locales')
+      })
+    })
+  })
+
+  describe('fetchAsset', () => {
+    describe('when the request is successful', () => {
+      beforeEach(() => {
+        nock(CMS_URL)
+          .get(
+            `/spaces/${mockSpace}/environments/${mockEnvironment}/assets/${mockId}/`
+          )
+          .reply(200, marketplaceHomepageBannerAssets[0])
+      })
+
+      it('should fetch and transform an asset to the correct format', async () => {
+        const result = await client.fetchAsset(
           mockSpace,
           mockEnvironment,
-          mockId,
-          mockContentType,
-          mockToken
+          mockId
         )
 
-        expect(result).toEqual(mockResponse)
+        expect(result.fields).toHaveProperty('title.en-US')
+        expect(result.fields).toHaveProperty('description.en-US')
+        expect(result.fields).toHaveProperty('file.en-US')
       })
     })
 
     describe('when the request fails', () => {
       beforeEach(() => {
-        global.fetch = jest.fn().mockResolvedValue({
-          ok: false
-        })
+        nock(CMS_URL)
+          .get(
+            `/spaces/${mockSpace}/environments/${mockEnvironment}/assets/${mockId}/`
+          )
+          .replyWithError('Failed to fetch asset data')
       })
 
       it('should throw an error', async () => {
         await expect(
-          client.fetchEntry(
-            mockSpace,
-            mockEnvironment,
-            mockId,
-            mockContentType,
-            mockToken
-          )
-        ).rejects.toThrow('Failed to fetch entity data')
+          client.fetchAsset(mockSpace, mockEnvironment, mockId)
+        ).rejects.toThrow('Failed to fetch asset data')
       })
     })
   })
+
+  describe('fetchAssetsFromEntryFields', () => {
+    describe('when there are fields to get the assets from', () => {
+      it('should get all assets referenced in the entry fields', async () => {
+        global.fetch = jest.fn().mockResolvedValue({
```

Comment:
This test is using a global fetch, would you mind replacing it with nock so all tests are the same? The nock should be defined in a `beforeEach`

Reply:
good catch